### PR TITLE
Feat: Cart context

### PR DESCRIPTION
**Pull request by Chakravarthi0**
implented cartcontext

### CODE REVIEW

File: src/components/cart-card/CartCard.jsx [hunk 1]
```diff
@@ -1,22 +1,20 @@
-import React, { useState } from "react";
+import React from "react";
+import { useCart } from "../../context/CartContext";
 import "./cart-card.css";
 
-function CartCard({ title, brand, imgSrc, newPrice, oldPrice, quantity }) {
-  const [chosenQuantity, setChosenQuantity] = useState(quantity || 1);
+function CartCard({ product }) {
+  const {
```
Comment (poornima-alugubelly) [inline src/components/cart-card/CartCard.jsx:6]: I like how u have destructured everything 👍

File: src/components/cart-card/CartCard.jsx [hunk 2]
```diff
@@ -26,41 +24,46 @@ function CartCard({ title, brand, imgSrc, newPrice, oldPrice, quantity }) {
         <p className="product-title">{title}</p>
         <p className="product-brand">{brand}</p>
         <div className="product-price">
-          <span className="new-price">₹{Intl.NumberFormat('en-IN').format(newPrice)}</span>
+          <span className="new-price">
+            ₹{Intl.NumberFormat("en-IN").format(newPrice)}
```
Comment (poornima-alugubelly) [inline src/components/cart-card/CartCard.jsx:28]: Number formatting is also very good here 👍

File: src/components/price-details/PriceDetails.jsx [hunk 3]
```diff
@@ -1,35 +1,70 @@
 import React from "react";
+import { useCart } from "../../context/index";
 import "./price-details.css";
 
 function PriceDetails() {
+  const { cartItems } = useCart();
+
+  const priceReducer = (acc, curr) => {
```
Comment (poornima-alugubelly) [inline src/components/price-details/PriceDetails.jsx:8]: May be u can make a separate utility for billing

**Review by poornima-alugubelly — APPROVED** [review]

**Review by mimansha-swarup — APPROVED** [review]
code is pretty clean
like how you are destructuring  at top only

**Pull request merged**